code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/impl/DeploymentManagerImpl.java

```diff
@@ -246,6 +246,17 @@ public void doWithLockNoResult() {
         });
     }
 
+    @Override
+    public void activateGlobalServicesForHost(long hostId) {
+        List<Service> services = sdSvc.getActiveGlobalServices();
+        for (Service service: services) {
```

Comment:
Just spoke with @ibuildthecloud about it - the problem is we currently don't have an ability to schedule non-blocking deploymentManager.serviceAcivate. Darren's healtcheck monitoring would also need this functionality for scheduling reconcile on heatlh state updte. We should think about what would be the best solution in this case.